tests/test_detect_regressions.py

```diff
@@ -0,0 +1,72 @@
+import pathlib
+
+import pandas
+import sqlalchemy
+
+
+def test_detect_regressions():
+
+    DB_FILE = pathlib.Path("./benchmark.db")
+    engine = sqlalchemy.create_engine(f"sqlite:///{DB_FILE}")
+
+    df = pandas.read_sql("select * from test_run where platform = 'linux'", engine)
+
+    # join runtime + py version
+    df = df.assign(
+        runtime=(
+            "coiled-"
+            + df.coiled_runtime_version
+            + "-py"
+            + df.python_version.str.split(".", n=2).str[:2].str.join(".")
+        ),
+        category=df.path.str.split("/", n=1).str[0],
+    )
+
+    # by_name_runtime = df.groupby(["name", "runtime"])
+    stats_dict = {}
+    regressions = []
+    runtimes = list(df.runtime.unique())
+    for runtime in runtimes:
+        by_test = df[(df.runtime == runtime)].groupby("name")
+
+        test_names = list(by_test.groups.keys())
+        for name in test_names:
+            df_test = by_test.get_group(name)
+
+            df_passed = df_test[df_test.call_outcome == "passed"]
+
+            # stats of latest 10 runs exclude last point
+
+            stats_dict[f"({runtime, name})"] = {
```

Comment:
I created a dictionary here because initially, I thought I might want to have all this info and dump it into a dataframe or something and we could do a table or something else for the dashboard. But I'm not sure it is the cleanest approach. Thoughts?